src/main/java/ch/njol/skript/expressions/ExprBeacon.java

```diff
@@ -0,0 +1,63 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.block.Beacon;
+import org.bukkit.block.Block;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+
+@Name("Beacon From Block")
+@Description("Gets a beacon type from a block")
+@Examples({
+	"set {_beacon} to beacon from (block at location(0,0,0))",
+	"set primary effect of {_beacon} to haste"
+})
+@Since("INSERT VERSION")
+public class ExprBeacon extends SimpleExpression<Beacon> {
```

Comment:
i'm not sure this should even exist, i've never seen this before. shouldn't BeaconValues just check if it's a beacon and return none if it's not a beacon?